### ISSUE TITLE
Some games require a long press on the option button

### ISSUE THREAD

**Issue by Sichroteph** [ayufan/steam-deck-tools#22]
Probably an isolated case, but the last ninja turtles game require to long press on the option button, hence disconnecting the controller.

**Comment by Sichroteph** [comment]
Red dead redemption 2 uses a long press too to display the map.

**Comment by ayufan** [comment]
Yeah. I saw that in some games. Lets see how annoying it is, and maybe someone will have suggestion to better mapping.

**Comment by marcoand75** [comment]
You can replace Options button with STEAM+Y that haven't binding active now
Options (hold for 1s) -> STEAM + Y  (hold for 1s)
Options (hold for 2s) -> STEAM + Y  (hold for 2s)

**Comment by ayufan** [comment]
@marcoand75 I'm not big fan of that as Valve might assign something to this later.

I was thinking about holding Options and Menu together. So, 3 horizontal lines + 2 boxes.

**Comment by marcoand75** [comment]
> @marcoand75 I'm not big fan of that as Valve might assign something to this later.
> 
> I was thinking about holding Options and Menu together. So, 3 horizontal lines + 2 boxes.
> 

It may be perfect, I suggest stream+y to fill the right functions...xyab
You can use steam+y maybe to map osd switch and for osd mode

**Comment by navarayuk** [comment]
> @marcoand75 I'm not big fan of that as Valve might assign something to this later.
> 
> I was thinking about holding Options and Menu together. So, 3 horizontal lines + 2 boxes.

Why not make it so users can edit it them selves, I like the 1 dots and option key and I also like the option and menu buttons option but not sure witch would feel nice, but I change to contoller mode before booting the game as some game don't like detecting controllers after they booted, and touchscreen is fine to boot a game.

Love your tool tho, SWICD for me worked but wouldn't let me map anything no matter what I tried I tried your tool set and worked right there and then. thanks for your hard work.

It does seem to be missing the 20 TDP optoin and a few little more tweaks but it's best I used so far for alot of the things.

**Comment by ayufan** [comment]
> Why not make it so users can edit it them selves, I like the 1 dots and option key and I also like the option and menu buttons option but not sure witch would feel nice

Here is longer explanation for the reasons: https://www.reddit.com/r/WindowsOnDeck/comments/z8nxak/opinionated_steam_controller/. Why we need to configure something where there could be a single convention?

Too much configuration is a hassle to user (to understand) but also hassle to maintain.

**Comment by navarayuk** [comment]
> > Why not make it so users can edit it them selves, I like the 1 dots and option key and I also like the option and menu buttons option but not sure witch would feel nice
> 
> Here is longer explanation for the reasons: https://www.reddit.com/r/WindowsOnDeck/comments/z8nxak/opinionated_steam_controller/. Why we need to configure something where there could be a single convention?
> 
> Too much configuration is a hassle to user (to understand) but also hassle to maintain.

Modualler options isn't a hassle you already have the framework for it, this option would fix all games in the future, but for me like I said I don't even see it as a problem as I change the mode before loading the game. If you said no I am leaving it fixed then so be it, but for me I want the single holding the button option so if you change it please add the change as a addon not fully removing the holding option key, it works fine for many of us.

**Comment by ayufan** [comment]
> Modualler options isn't a hassle you already have the framework for it, this option would fix all games in the future, but for me like I said I don't even see it as a problem as I change the mode before loading the game

Every single added configuration is a hassle to support, explain and maintain:

- It is not about the framework.
- Each configuration needs to be tested, and can break overtime.
- You need to expose many options.
- You need to document them.
- You need to make users aware of it.
- Different users might have different configuration.
- When users help each other they will give non working solutions.
 
I will look for more feedback and will change it based on this feedback, but it will not be configurable. It will be changed for all. My preference now is `3 horizontal lines + 2 boxes`, but will wait for more feedback as I want to make this change once.

The option can also be:

- if you are in desktop: you hold existing 3 lines to switch into controller
- if you are in controller: you need to hold two buttons to switch to next or go back to desktop

**Comment by navarayuk** [comment]
> > Modualler options isn't a hassle you already have the framework for it, this option would fix all games in the future, but for me like I said I don't even see it as a problem as I change the mode before loading the game
> 
> Every single added configuration is a hassle to support, explain and maintain:
> 
> * It is not about the framework.
> * Each configuration needs to be tested, and can break overtime.
> * You need to expose many options.
> * You need to document them.
> * You need to make users aware of it.
> * Different users might have different configuration.
> * When users help each other they will give non working solutions.
> 
> I will look for more feedback and will change it based on this feedback, but it will not be configurable. It will be changed for all. My preference now is `3 horizontal lines + 2 boxes`, but will wait for more feedback as I want to make this change once.
> 
> The option can also be:
> 
> * if you are in desktop: you hold existing 3 lines to switch into controller
> * if you are in controller: you need to hold two buttons to switch to next or go back to desktop

Though I don't agree with the framework comment we all program different but that last option would work, keep the timers on it also and I think it would work well, I might give your source a look later see if I can get my head around it I don't even know what lanuage it's in right now :D 

I do agree users could get confused and mess things up if two many options but key bindings are easy to test and options to change them also helps people that might not even be able to access them buttoms. I understand where your coming from but I really think closing the door on creating custom bindings for shortcuts in the tool is a option people like myself would love.

Granted the options you posted are also good work around to the problem people are having now but ya it's just feedback.

**Comment by ayufan** [comment]
> I understand where your coming from but I really think closing the door on creating custom bindings for shortcuts in the tool is a option people like myself would love.

I'm planning to offer "dynamic, but programmable" profiles to allow you to model your controllers. This is example implementing turbo buttons: https://github.com/ayufan/steam-deck-tools/blob/develop/SteamController/UserProfiles/Turbo.x360.cs

> Granted the options you posted are also good work around to the problem people are having now but ya it's just feedback.

Thanks. We will see where the discussion goes.

**Comment by ayufan** [comment]
> is macro options some games what need a few keyboard inputs like lets say minecraft for /sethome and /home this could be mapped to a macro on R4/L4R5/L5

All of that is already possible via `UserProfiles` above. You can generate any type of inputs. You can control mouse, etc. You could also consume accell or gyro data in your own way. Here's the current idea of API interface exposed: https://github.com/ayufan/steam-deck-tools/blob/develop/SteamController/Profiles/Dynamic/Globals.cs

**Comment by julolosvalles26** [comment]
In assassins, you also have to press that button to continue the mission, but you can search the list of missions without the button, taking into account that you make the most complete program for Windows and that it is very easy to put the remote on and off, so it's not so bad.

**Comment by marcoand75** [comment]
@ayufan regarding this issue like you make changes for custom TDP, i think the better and easy way is to have in ini file these two parameters
![image](https://user-images.githubusercontent.com/4765145/217558792-26422e42-ffe1-4cc6-9765-95dcf1b673a1.png)

**Comment by ayufan** [comment]
Please no. I will make workflow change in 0.7.x.

**Comment by ydq1** [comment]
> @ayufan regarding this issue like you make changes for custom TDP, i think the better and easy way is to have in ini file these two parameters. obviously is my opinion. i change myself and recompiled with these settings and i solved.
> 
> ![image](https://user-images.githubusercontent.com/4765145/217558792-26422e42-ffe1-4cc6-9765-95dcf1b673a1.png)
> 

Could you share ?

**Comment by marcoand75** [comment]
In ShortcutsProfile.cs
![image](https://user-images.githubusercontent.com/4765145/222134919-49fd34df-bc3f-48a2-be11-20f21fc8532d.png)

change values:
private readonly TimeSpan HoldToSwitchProfile = TimeSpan.FromSeconds(3);
private readonly TimeSpan HoldToSwitchDesktop = TimeSpan.FromSeconds(4);

recompile and use

**Comment by ydq1** [comment]
Thanks!!!

**Comment by ydq1** [comment]
- - It works. I changed as holding 20s ....~~~~
